airbyte-ci/connectors/live-tests/src/live_tests/commons/backends/file_backend.py

```diff
@@ -22,101 +30,57 @@ class FileBackend(BaseBackend):
-
-            elif message.type == AirbyteMessageType.RECORD:
-                assert isinstance(messages_by_type["records"], dict)
-                messages_by_type["records"][message.record.stream].append(message.record)
-
-            elif message.type == AirbyteMessageType.SPEC:
-                assert isinstance(messages_by_type["spec"], list)
-                messages_by_type["spec"].append(message.spec)
-
-            elif message.type == AirbyteMessageType.STATE:
-                if message.state.stream and message.state.stream.stream_descriptor:
-                    stream_name = message.state.stream.stream_descriptor.name
-                    stream_namespace = message.state.stream.stream_descriptor.namespace
-                    key = f"{stream_name}_{stream_namespace}" if stream_namespace else stream_name
-                else:
-                    key = "_global_states"
-                assert isinstance(messages_by_type["states"], dict)
-                messages_by_type["states"][key].append(message.state)
-            elif message.type == AirbyteMessageType.TRACE:
-                assert isinstance(messages_by_type["traces"], list)
-                messages_by_type["traces"].append(message.trace)
-            elif message.type == AirbyteMessageType.LOG:
-                assert isinstance(messages_by_type["logs"], list)
-                messages_by_type["logs"].append(message.log)
-            elif message.type == AirbyteMessageType.CONTROL:
-                assert isinstance(messages_by_type["controls"], list)
-                messages_by_type["controls"].append(message.control)
-        return messages_by_type
+    async def write(self, airbyte_messages: Iterable[AirbyteMessage]):
+        @cached(cache=FileDescriptorLRUCache(maxsize=250))
+        def _open_file(path: Path) -> TextIO:
+            return open(path, "a")
+
+        try:
+            for _message in airbyte_messages:
+                if not isinstance(_message, AirbyteMessage):
+                    continue
+                filepath, message = self._get_filepath_and_message(_message)
+                _open_file(self._output_directory / filepath).write(f"{message}\n")
+        finally:
+            for f in _open_file.cache.values():
+                f.close()
+
+    def _get_filepath_and_message(self, message: AirbyteMessage) -> Tuple[str, str]:
+        if message.type == AirbyteMessageType.CATALOG:
+            return self.RELATIVE_CATALOGS_PATH, message.catalog.json()
+
+        elif message.type == AirbyteMessageType.CONNECTION_STATUS:
+            return self.RELATIVE_CONNECTION_STATUS_PATH, message.connectionStatus.json()
+
+        elif message.type == AirbyteMessageType.RECORD:
+            record = json.loads(message.record.json())
+            # TODO: once we have a comparator and/or database backend implemented we can remove this
+            record.pop("emitted_at", None)
```

Comment:
nit: can we store a list of key paths to pop as a class attribute, with `dpath` / `pydash`?

Reply:
👍 done